docs/tools.md

```diff
@@ -90,3 +90,19 @@ El primero es que el estado de mantenimiento de dotenv es mucho mejor que el de
 El segundo es que, mientras que dotenv no tiene dependencias, nconf tiene 4. Cuantas menos dependencias la librería mejor porque menos dependencias tendrá el proyecto.
 El tercero es que dotenv tiene el soporte para tipos totalmente integrado mientras que nconfg las tiene en un paquete a parte(esto es un detalle).
 
+
+## REST API Framework
+
+Requisistos:
+1. Flexible: Llegado este punto el proyecto ya está estructurado. El framework ha de tener la capacidad de adaptarse a la estructura actual del proyecto.
+2. Eficiente: Ha de ser capaz de gestionar suficientes peticiones por unidad de tiempo.
+3. Integrable: Fundamentalmente con las herramientas escogidas para test y logging
+4. Soporte para tipos de typescript
+
+Se han considerado los siguientes frameworks: [nest](https://www.npmjs.com/package/@nestjs/core), [fastify](https://www.npmjs.com/package/fastify), [hapi](https://www.npmjs.com/package/@hapi/hapi), [restify](https://www.npmjs.com/package/restify).
+
+- Nest es la alternativa más popular. Nest es un proyecto apoyado por grandes multinacionales como Redhat y empresas como NX y Labster que tiene como objetivo resolver los problemas arquitectónicos frecuentes en las aplicaciones escritas en javascript/typescript: "... However, while plenty of superb libraries, helpers, and tools exist for Node (and server-side JavaScript), none of them effectively solve the main problem of - Architecture". Esto implica que las herramientas necesarias para construir el API están pensadas para usarse con el resto de las herramientas del proyecto pero sobre todo que hay que seguir la estructura de aplicación propuesta. Por lo tanto no verifica la tercera condición y ha sido descartado.
+- Fastify es la segunda alternativa más popular y la más novedosa. Esta librería esta centrada en la eficiencia siendo esta uno de sus principales atractivos; es la alternativa más eficiente de las consideradas según sus propios benchmark que se pueden encontrar en https://github.com/fastify/benchmarks. Su estado de mantenimiento es muy bueno y se integra bien con el logger elegido pero se integra muy mal con la librería de test luego no termina de verificar tercera condición. Ha sido descartada por este motivo y porque su novedad implica una falta de pluggins disponibles y de posible documentación de cara a la implementación de características sofisticadas.
```

Comment:
Sorry por no haber comentado nada, cosas de tomarse el día 1 de fiesta. Como ya lo has puesto para revisión, solo te diré como comentario que fastify tiene la funcion "inject" para hacer peticiones HTTP "fake" en los test, con JEST funciona bastante bien.

Reply:
> Sorry por no haber comentado nada, cosas de tomarse el día 1 de fiesta. Como ya lo has puesto para revisión, solo te diré como comentario que fastify tiene la funcion "inject" para hacer peticiones HTTP "fake" en los test, con JEST funciona bastante bien, ya no sé con Mocha, pero supongo será 3/4 de lo mismo.

Aunque lo haya puesto para revisión podéis comentar lo que queráis, si me da tiempo a cambiarlo antes de la revisión, lo haré. Tienes razón con el tema de los injects, lo que realmente no me gusta es que la declaración de las rutas y sus llamadas sean distintas(entre otras cosas). Voy a actualizar la documentación.

Reply:
Hecho 6db2eba.